src/NCronJob/Registry/JobRegistry.cs

```diff
@@ -4,58 +4,45 @@ namespace NCronJob;
 
 internal sealed class JobRegistry
 {
-    private readonly HashSet<JobDefinition> allJobs;
-    private readonly ImmutableHashSet<JobDefinition> oneTimeJobs;
-    private readonly HashSet<JobDefinition> cronJobs;
+    private readonly DontKnowHowToNameThat storage;
 
-    public JobRegistry(IEnumerable<JobDefinition> jobs)
-    {
-        var jobDefinitions = jobs as JobDefinition[] ?? jobs.ToArray();
-        allJobs = new HashSet<JobDefinition>(jobDefinitions, JobDefinitionEqualityComparer.Instance);
-        cronJobs = new HashSet<JobDefinition>(jobDefinitions.Where(c => c.CronExpression is not null), JobDefinitionEqualityComparer.Instance);
-        oneTimeJobs = [..jobDefinitions.Where(c => c.IsStartupJob)];
+    private HashSet<JobDefinition> AllJobs => new(storage.JobDefinitions, JobDefinitionEqualityComparer.Instance);
```

Comment:
This is ugly. I know. :wink:

Reply:
However, I haven't found a better option yet...

Thoughts?

Reply:
Given that `JobDefinition` is a `record` - shouldn't that automatically work and if not, providing a custom `GetHashCode` and `Equals` method?

Reply:
Ok. I've managed to stop recreating a new HashSet every time the property is accessed. That required a slight change to the comparer in order to not deduplicate dynamic jobs.